bridge-token/src/lib.rs

```diff
@@ -43,72 +47,86 @@ impl BridgeToken {
         assert!(!env::state_exists(), "Already initialized");
         Self {
             controller: env::predecessor_account_id(),
-            token: Token::new(env::predecessor_account_id(), 0u128),
+            token: FungibleToken::new(b"t".to_vec()),
+            name: String::default(),
+            symbol: String::default(),
+            reference: String::default(),
+            reference_hash: Base64VecU8(vec![]),
+            decimals: 0,
             paused: Mask::default(),
         }
     }
 
+    pub fn set_metadata(
+        &mut self,
+        name: Option<String>,
+        symbol: Option<String>,
+        reference: Option<String>,
+        reference_hash: Option<Base64VecU8>,
+        decimals: Option<u8>,
+    ) {
+        // Only owner can change the metadata
+        assert_eq!(env::current_account_id(), env::signer_account_id());
```

Comment:
Doesn't using `signer_account_id` have the same problems as using `tx.origin` in Solidity? Please change to `predecessor_account_id`.

Reply:
Solved in https://github.com/near/rainbow-token-connector/pull/39/commits/7482c5d67b895380b11695b597b0174543484a9d

Moreover same issue existed with current implementation of `AdminControlled` which is fixed here: https://github.com/near/rainbow-bridge/pull/524. It is already fixed, and added PR to use respective version: https://github.com/near/rainbow-token-connector/pull/39/commits/dbc227bc4881721e9201d731ec54d81618e36003